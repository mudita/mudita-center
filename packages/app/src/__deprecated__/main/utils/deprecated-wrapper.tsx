/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { FunctionComponent } from "react"

type Element = FunctionComponent<any>

const deprecated = (component: Element, message?: string): Element => {
  if (process.env.NODE_ENV === "development") {
    const basicMessage = `This component is deprecated.`
    if (message) {
      console.warn(`${basicMessage} ${message}`)
    } else {
      console.warn(basicMessage)
    }
  }

  return component
}

export default deprecated
