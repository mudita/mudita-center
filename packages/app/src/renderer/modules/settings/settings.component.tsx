/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import React, { ComponentProps } from "react"
import { FunctionComponent } from "Renderer/types/function-component.interface"
import SettingsUI from "Renderer/components/rest/settings/settings-ui.component"

type Properties = ComponentProps<typeof SettingsUI>

const Settings: FunctionComponent<Properties> = ({
  ...uiComponentProperties
}) => {
  return <SettingsUI {...uiComponentProperties} />
}

export default Settings
