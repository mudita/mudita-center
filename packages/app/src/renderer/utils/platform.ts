/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

export class Platform {
  public macOs(): boolean {
    return process.platform === "darwin"
  }

  public linux(): boolean {
    return process.platform === "linux"
  }

  public windows(): boolean {
    return process.platform === "win32"
  }
}

export const platform = new Platform()
