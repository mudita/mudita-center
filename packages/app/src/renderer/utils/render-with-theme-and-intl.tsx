/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import {
  Queries,
  render,
  RenderOptions,
  RenderResult,
} from "@testing-library/react"
import React from "react"
import { IntlProvider } from "react-intl"
import translationConfig from "App/translations.config.json"
import localeEn from "Renderer/locales/default/en-US.json"
import { ThemeProvider } from "styled-components"
import theme from "../styles/theming/theme"
import extractLanguageKeys from "Renderer/utils/extract-test-locale"
import ReactModal from "react-modal"

ReactModal.setAppElement(document.createElement("div"))
const testLocale = extractLanguageKeys(localeEn)

export function renderWithThemeAndIntl(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q>
): RenderResult {
  return render(
    <ThemeProvider theme={theme}>
      <IntlProvider
        defaultLocale={translationConfig.defaultLanguage}
        locale={translationConfig.defaultLanguage}
        messages={process.env.NODE_ENV === "test" ? testLocale : localeEn}
      >
        {ui}
      </IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}
