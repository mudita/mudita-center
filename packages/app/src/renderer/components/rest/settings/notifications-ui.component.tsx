/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/LICENSE.md
 */

import { FunctionComponent } from "Renderer/types/function-component.interface"
import { TextDisplayStyle } from "Renderer/components/core/text/text.component"
import { FormattedMessage } from "react-intl"
import { ActionsWrapper } from "Renderer/components/rest/messages/threads-table.component"
import React from "react"
import {
  Data,
  SettingsDescription,
  SettingsDescriptionWrapper,
  SettingsLabel,
  SettingsTableRow,
  SettingsWrapper,
} from "Renderer/components/rest/settings/settings-ui.component"
import SettingsToggler from "Renderer/components/rest/settings/settings-toggler.component"
import { noop } from "Renderer/utils/noop"

interface Props {
  appIncomingCalls?: boolean
  appIncomingMessages?: boolean
  appLowBattery?: boolean
  appOsUpdates?: boolean
  setIncomingCalls?: (option: boolean) => void
  setIncomingMessages?: (option: boolean) => void
  setLowBattery?: (option: boolean) => void
  setOsUpdates?: (option: boolean) => void
}

const NotificationsUI: FunctionComponent<Props> = ({
  appIncomingCalls,
  setIncomingCalls = noop,
  appIncomingMessages,
  setIncomingMessages = noop,
  appLowBattery,
  setLowBattery = noop,
  appOsUpdates,
  setOsUpdates = noop,
}) => {
  return (
    <SettingsWrapper>
      <SettingsDescriptionWrapper>
        <SettingsDescription
          displayStyle={TextDisplayStyle.MediumFadedLightText}
        >
          <FormattedMessage id="view.name.settings.notifications.description" />
        </SettingsDescription>
      </SettingsDescriptionWrapper>
      <SettingsTableRow checkMode={false}>
        <Data>
          <SettingsLabel displayStyle={TextDisplayStyle.LargeText}>
            <FormattedMessage id="view.name.settings.notifications.incomingCallsNotificationsLabel" />
          </SettingsLabel>
        </Data>
        <ActionsWrapper>
          <SettingsToggler
            toggleValue={appIncomingCalls}
            onToggle={setIncomingCalls}
          />
        </ActionsWrapper>
      </SettingsTableRow>
      <SettingsTableRow checkMode={false}>
        <Data>
          <SettingsLabel displayStyle={TextDisplayStyle.LargeText}>
            <FormattedMessage id="view.name.settings.notifications.incomingMessagesNotifications" />
          </SettingsLabel>
        </Data>
        <ActionsWrapper>
          <SettingsToggler
            toggleValue={appIncomingMessages}
            onToggle={setIncomingMessages}
          />
        </ActionsWrapper>
      </SettingsTableRow>
      <SettingsTableRow checkMode={false}>
        <Data>
          <SettingsLabel displayStyle={TextDisplayStyle.LargeText}>
            <FormattedMessage id="view.name.settings.notifications.lowBatteryNotifications" />
          </SettingsLabel>
        </Data>
        <ActionsWrapper>
          <SettingsToggler
            toggleValue={appLowBattery}
            onToggle={setLowBattery}
          />
        </ActionsWrapper>
      </SettingsTableRow>
      <SettingsTableRow checkMode={false}>
        <Data>
          <SettingsLabel displayStyle={TextDisplayStyle.LargeText}>
            <FormattedMessage id="view.name.settings.notifications.pureOsUpdatesNotifications" />
          </SettingsLabel>
        </Data>
        <ActionsWrapper>
          <SettingsToggler toggleValue={appOsUpdates} onToggle={setOsUpdates} />
        </ActionsWrapper>
      </SettingsTableRow>
    </SettingsWrapper>
  )
}

export default NotificationsUI
