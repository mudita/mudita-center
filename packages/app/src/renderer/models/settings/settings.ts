/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { version } from "../../../../package.json"
import isVersionGreater from "App/overview/helpers/is-version-greater"
import { RootState } from "Renderer/store"
import {
  getAppSettings,
  updateAppSettings,
} from "Renderer/requests/app-settings.request"
import {
  AppSettings,
  SettingsUpdateOption,
  SettingsState,
} from "App/main/store/settings.interface"
import { createSelector, Slicer, StoreSelectors } from "@rematch/select"
import { createModel } from "@rematch/core"
import { RootModel } from "Renderer/models/models"
import logger from "App/main/utils/logger"
import e2eSettings from "Renderer/models/settings/e2e-settings.json"
import { isToday } from "Renderer/utils/is-today"
import getDeviceLogs from "Renderer/requests/get-device-logs.request"
import sendDiagnosticDataRequest from "Renderer/requests/send-diagnostic-data.request"
import { DeviceResponseStatus } from "Backend/adapters/device-response.interface"
import getApplicationConfiguration from "App/renderer/requests/get-application-configuration.request"

const simulatePhoneConnectionEnabled = process.env.simulatePhoneConnection

export const initialState: SettingsState = {
  appUpdateAvailable: undefined,
  lowestSupportedOsVersion: undefined,
  lowestSupportedCenterVersion: undefined,
  appCurrentVersion: version,
  appUpdateStepModalDisplayed: false,
  settingsLoaded: false,
  appUpdateRequired: false,
  appLatestVersion: "",
}

const settings = createModel<RootModel>({
  state: initialState,
  reducers: {
    update(state: SettingsState, payload: Partial<SettingsState>) {
      return { ...state, ...payload }
    },
  },
  effects: (d) => {
    const dispatch = (d as unknown) as RootState

    return {
      async loadSettings() {
        const appSettings = await getAppSettings()
        const applicationConfiguration = await getApplicationConfiguration()

        let appUpdateRequired = false

        try {
          appUpdateRequired = isVersionGreater(
            applicationConfiguration.centerVersion,
            version
          )
        } catch (error) {
          logger.error(
            `Settings -> LoadSettings: Check that app update required fails: ${error.message}`
          )
        }

        const settings = {
          ...appSettings,
          appUpdateRequired,
          lowestSupportedOsVersion: applicationConfiguration.osVersion,
          settingsLoaded: true,
        }

        if (simulatePhoneConnectionEnabled) {
          dispatch.settings.update({
            ...e2eSettings,
            ...settings,
          })
        } else {
          dispatch.settings.update(settings)
        }

        appSettings.appCollectingData
          ? logger.enableRollbar()
          : logger.disableRollbar()
      },
      async updateSettings(option: SettingsUpdateOption) {
        await updateAppSettings(option)
        dispatch.settings.update({ [option.key]: option.value })
      },
      setTethering(value: AppSettings["appTethering"]) {
        this.updateSettings({ key: "appTethering", value })
      },
      setIncomingCalls(value: AppSettings["appIncomingCalls"]) {
        this.updateSettings({ key: "appIncomingCalls", value })
      },
      setIncomingMessages(value: AppSettings["appIncomingMessages"]) {
        this.updateSettings({ key: "appIncomingMessages", value })
      },
      setLowBattery(value: AppSettings["appLowBattery"]) {
        this.updateSettings({ key: "appLowBattery", value })
      },
      setOsUpdates(value: AppSettings["appOsUpdates"]) {
        this.updateSettings({ key: "appOsUpdates", value })
      },
      setNonStandardAudioFilesConversion(
        value: AppSettings["appNonStandardAudioFilesConversion"]
      ) {
        this.updateSettings({
          key: "appNonStandardAudioFilesConversion",
          value,
        })
      },
      setConvert(value: AppSettings["appConvert"]) {
        this.updateSettings({ key: "appConvert", value })
      },
      setConversionFormat(value: AppSettings["appConversionFormat"]) {
        this.updateSettings({ key: "appConversionFormat", value })
      },
      setAppTray(value: AppSettings["appTray"]) {
        this.updateSettings({ key: "appTray", value })
      },
      setPureOsBackupLocation(value: AppSettings["pureOsBackupLocation"]) {
        this.updateSettings({ key: "pureOsBackupLocation", value })
      },
      setPureOsDownloadLocation(value: AppSettings["pureOsDownloadLocation"]) {
        this.updateSettings({ key: "pureOsDownloadLocation", value })
      },
      setLanguage(value: AppSettings["language"]) {
        this.updateSettings({ key: "language", value })
      },
      setAppUpdateStepModalDisplayed() {
        dispatch.settings.update({ appUpdateStepModalDisplayed: true })
      },
      setDiagnosticSentTimestamp(
        value: AppSettings["diagnosticSentTimestamp"]
      ) {
        this.updateSettings({ key: "diagnosticSentTimestamp", value })
      },
      toggleAppCollectingData(value: boolean) {
        this.updateSettings({ key: "appCollectingData", value })
        value ? logger.enableRollbar() : logger.disableRollbar()
      },
      toggleAppUpdateAvailable(appUpdateAvailable: boolean) {
        dispatch.settings.update({ appUpdateAvailable })
      },
      async sendDiagnosticData(_, state): Promise<void> {
        const { appCollectingData, diagnosticSentTimestamp } = state.settings
        const { serialNumber } = state.basicInfo

        if (serialNumber === undefined) {
          logger.error(
            `Send Diagnostic Data: device logs fail. SerialNumber is undefined.`
          )
          return
        }
        if (!appCollectingData) {
          logger.info("Send Diagnostic Data: user no allowed sent data")
          return
        }

        if (isToday(new Date(diagnosticSentTimestamp))) {
          logger.info(
            `Send Diagnostic Data: data was sent at ${diagnosticSentTimestamp}`
          )
          return
        }
        const { status, data = "", error } = await getDeviceLogs()
        if (status !== DeviceResponseStatus.Ok) {
          logger.error(
            `Send Diagnostic Data: device logs fail. Message: ${error?.message}.`
          )
          return
        }

        try {
          const { status } = await sendDiagnosticDataRequest(data, serialNumber)
          if (status !== 200) {
            logger.error(
              `Send Diagnostic Data: send diagnostic data request. Status: ${status}`
            )
          }
          const nowTimestamp = Date.now()
          await dispatch.settings.setDiagnosticSentTimestamp(nowTimestamp)
          logger.info(
            `Send Diagnostic Data: data was sent successfully at ${nowTimestamp}, serialNumber: ${serialNumber}`
          )
        } catch {
          logger.error(`Send Diagnostic Data: send diagnostic data request.`)
        }
      },
      setAppLatestVersion(appLatestVersion: string) {
        dispatch.settings.update({ appLatestVersion })
      },
    }
  },
  selectors: (slice: Slicer<SettingsState>) => ({
    appCollectingData() {
      return slice(({ appCollectingData }) => appCollectingData)
    },
    appUpdateStepModalDisplayed() {
      return slice(
        ({ appUpdateStepModalDisplayed }) => appUpdateStepModalDisplayed
      )
    },
    settingsLoaded() {
      return slice(({ settingsLoaded }) => settingsLoaded)
    },
    initialModalsShowed(models: StoreSelectors<any>) {
      return createSelector(
        models.settings.appCollectingData,
        models.settings.appUpdateStepModalDisplayed,
        models.settings.settingsLoaded,
        (appCollectingData, appUpdateStepModalDisplayed, settingsLoaded) => {
          return (
            settingsLoaded &&
            appUpdateStepModalDisplayed &&
            appCollectingData !== undefined
          )
        }
      )
    },
  }),
})

export default settings
