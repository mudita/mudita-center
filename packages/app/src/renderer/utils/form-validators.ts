/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { RegisterOptions } from "react-hook-form/dist/types"
import { intl } from "Renderer/utils/intl"

export const phoneNumberRegexp = /^(\+?)(\d(\s?\d)+)$/im

export const primaryPhoneNumberValidator = (
  fields: Record<any, any>
): RegisterOptions => ({
  minLength: {
    value: 5,
    message: intl.formatMessage(
      { id: "component.formErrorTooShort" },
      { minLength: 5 }
    ),
  },
  maxLength: {
    value: 15,
    message: intl.formatMessage(
      { id: "component.formErrorTooLong" },
      { maxLength: 15 }
    ),
  },
  pattern: {
    value: phoneNumberRegexp,
    message: intl.formatMessage({
      id: "component.formErrorNumbersAndSpacesOnly",
    }),
  },
  validate: (value): string | undefined => {
    if (value === fields.secondaryPhoneNumber) {
      return intl.formatMessage({
        id: "component.formErrorNumberUnique",
      })
    }

    return
  },
})

export const secondaryPhoneNumberValidator = (
  fields: Record<any, any>
): RegisterOptions => ({
  minLength: {
    value: 5,
    message: intl.formatMessage(
      { id: "component.formErrorTooShort" },
      { minLength: 5 }
    ),
  },
  maxLength: {
    value: 15,
    message: intl.formatMessage(
      { id: "component.formErrorTooLong" },
      { maxLength: 15 }
    ),
  },
  pattern: {
    value: phoneNumberRegexp,
    message: intl.formatMessage({
      id: "component.formErrorNumbersAndSpacesOnly",
    }),
  },
  validate: (value): string | undefined => {
    if (value === fields.primaryPhoneNumber) {
      return intl.formatMessage({
        id: "component.formErrorNumberUnique",
      })
    }

    return
  },
})

export const emailValidator: RegisterOptions = {
  pattern: {
    value:
      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
    message: intl.formatMessage({ id: "component.formErrorInvalidEmail" }),
  },
  required: intl.formatMessage({ id: "component.formErrorRequiredEmail" }),
}
