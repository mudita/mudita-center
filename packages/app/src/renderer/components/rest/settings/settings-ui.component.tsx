/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { AppSettings } from "App/main/store/settings.interface"
import React from "react"
import { FunctionComponent } from "Renderer/types/function-component.interface"
import {
  ActionsWrapper,
  Name,
  TableRow,
} from "Renderer/components/rest/messages/threads-table.component"
import Text, {
  TextDisplayStyle,
} from "Renderer/components/core/text/text.component"
import { noop } from "Renderer/utils/noop"
import styled from "styled-components"
import { FormattedMessage } from "react-intl"
import { borderColor } from "Renderer/styles/theming/theme-getters"
import SettingsToggler from "Renderer/components/rest/settings/settings-toggler.component"
import { SettingsTestIds } from "Renderer/modules/settings/settings.enum"

export const SettingsTableRow = styled(TableRow)`
  grid-template-areas: "Checkbox Actions";
  grid-template-columns: 1fr 15rem;
  border-bottom: solid 0.1rem ${borderColor("list")};
  &:hover {
    background-color: transparent;
  }
`

export const Data = styled.div`
  grid-area: Checkbox;
  align-self: center;
`

export const SettingsLabel = styled(Name)`
  margin-left: 4rem;
`

export const SettingsDescriptionWrapper = styled.div`
  border-bottom: solid 0.1rem ${borderColor("list")};
`

export const SettingsDescription = styled(Text)`
  margin-left: 4rem;
  margin-bottom: 3.2rem;
`

export const SettingsWrapper = styled.section``

interface Properties {
  appAutostart: boolean
  appTethering: boolean
  appCollectingData: boolean
  setTethering?: (option: AppSettings["appTethering"]) => void
  setCollectingData?: (option: AppSettings["appCollectingData"]) => void
}

const SettingsUI: FunctionComponent<Properties> = ({
  appAutostart,
  appTethering,
  setTethering = noop,
  appCollectingData,
  setCollectingData = noop,
}) => {
  return (
    <SettingsWrapper data-testid={SettingsTestIds.Wrapper}>
      {/*TODO: Remove condition below when tethering will be available on phone*/}
      {process.env.NODE_ENV === "development" && (
        <SettingsTableRow
          checkMode={false}
          data-testid={SettingsTestIds.TableRow}
        >
          <Data>
            <SettingsLabel displayStyle={TextDisplayStyle.LargeText}>
              <FormattedMessage id="module.settings.tetheringLabel" />
            </SettingsLabel>
          </Data>
          <ActionsWrapper>
            <SettingsToggler
              toggleValue={appTethering}
              onToggle={setTethering}
            />
          </ActionsWrapper>
        </SettingsTableRow>
      )}
      <SettingsTableRow
        checkMode={false}
        data-testid={SettingsTestIds.TableRow}
      >
        <Data>
          <SettingsLabel displayStyle={TextDisplayStyle.LargeText}>
            <FormattedMessage id="module.settings.collectingData" />
          </SettingsLabel>
        </Data>
        <ActionsWrapper>
          <SettingsToggler
            toggleValue={
              appCollectingData === undefined ? false : appCollectingData
            }
            onToggle={setCollectingData}
          />
        </ActionsWrapper>
      </SettingsTableRow>
    </SettingsWrapper>
  )
}

export default SettingsUI
