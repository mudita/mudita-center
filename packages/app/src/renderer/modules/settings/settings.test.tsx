/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/LICENSE.md
 */

import { renderWithThemeAndIntl } from "Renderer/utils/render-with-theme-and-intl"
import "@testing-library/jest-dom/extend-expect"
import React from "react"
import SettingsUI from "Renderer/components/rest/settings/settings-ui.component"
import { SettingsTestIds } from "Renderer/modules/settings/settings.enum"

const renderer = (
  config = {
    appAutostart: false,
    appTethering: false,
    appCollectingData: false,
  }
) => renderWithThemeAndIntl(<SettingsUI {...config} />)

test("matches snapshot", () => {
  const { container } = renderer()
  expect(container).toMatchSnapshot()
})

test("renders wrapper properly", () => {
  const { queryByTestId } = renderer()
  expect(queryByTestId(SettingsTestIds.Wrapper)).toBeInTheDocument()
})

test("renders description properly", () => {
  const { queryByTestId } = renderer()
  expect(queryByTestId(SettingsTestIds.Description)).toBeInTheDocument()
})

test("renders at least one table row", () => {
  const { queryAllByTestId } = renderer()
  expect(
    queryAllByTestId(SettingsTestIds.TableRow).length
  ).toBeGreaterThanOrEqual(1)
})
