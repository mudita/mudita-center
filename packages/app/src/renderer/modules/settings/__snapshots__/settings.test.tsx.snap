// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
.c3 {
  margin: 0;
  font-size: 1.6rem;
  line-height: 1.05;
}

.c12 {
  margin: 0;
  font-size: 1.2rem;
  line-height: 1.2;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

.c4 {
  grid-area: Name;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
  margin-bottom: 0.4rem;
}

.c6 {
  grid-area: Actions;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 9rem;
}

.c0 {
  display: grid;
  grid-template-columns: 11rem 1fr 9rem;
  grid-template-areas: "Checkbox . Actions";
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  height: 9rem;
  min-height: 9rem;
  box-sizing: border-box;
  border-bottom: solid 0.1rem #d2d6db;
  -webkit-transition: background-color 0.15s ease-in-out;
  transition: background-color 0.15s ease-in-out;
}

.c0:hover {
  background-color: #f4f5f6;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  width: 17rem;
  height: 4rem;
  color: #ffffff;
  border-radius: 0.5rem;
  background: #3b3f42;
  border: 0.1rem solid #3b3f42;
}

.c9:hover {
  background: #232426;
}

.c9 g {
  fill: #ffffff;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  width: 17rem;
  height: 4rem;
  border-radius: 0.5rem;
  border: 0.1rem solid #3b3f42;
}

.c14 g {
  fill: #000000;
}

.c14:hover {
  border-color: #000000;
}

.c13 {
  white-space: nowrap;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
}

.c10 {
  --radius: 0.5rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  z-index: 1;
  pointer-events: all;
  z-index: 3;
}

.c10:disabled {
  pointer-events: none;
}

.c10:hover {
  z-index: 2;
}

.c10:not(:first-of-type) {
  border-radius: 0;
  margin-left: -0.1rem;
}

.c10:first-of-type:not(:last-of-type) {
  border-radius: var(--radius) 0 0 var(--radius);
}

.c10:last-of-type:not(:first-of-type) {
  border-radius: 0 var(--radius) var(--radius) 0;
}

.c15 {
  --radius: 0.5rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  z-index: 1;
  pointer-events: all;
}

.c15:disabled {
  pointer-events: none;
}

.c15:hover {
  z-index: 2;
}

.c15:not(:hover) {
  border: 0.1rem solid #cdcecf;
  color: #6a6a6a;
}

.c15:not(:hover) g {
  fill: #cdcecf;
}

.c15:not(:first-of-type) {
  border-radius: 0;
  margin-left: -0.1rem;
}

.c15:first-of-type:not(:last-of-type) {
  border-radius: var(--radius) 0 0 var(--radius);
}

.c15:last-of-type:not(:first-of-type) {
  border-radius: 0 var(--radius) var(--radius) 0;
}

.c8 {
  margin-right: 4rem;
}

.c11 {
  padding: 0 3.6rem;
  width: 75%;
}

.c1 {
  grid-template-areas: "Checkbox Actions";
  grid-template-columns: 1fr 15rem;
  border-bottom: solid 0.1rem #d2d6db;
}

.c1:hover {
  background-color: transparent;
}

.c2 {
  grid-area: Checkbox;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c5 {
  margin-left: 4rem;
}

<div>
  <section
    class=""
    data-testid="settings-wrapper"
  >
    <div
      class="c0 c1"
      data-testid="settings-tablerow"
    >
      <div
        class="c2"
      >
        <p
          class="c3 c4 c5"
        >
          [value] module.settings.collectingData
        </p>
      </div>
      <div
        class="c6"
      >
        <section
          class="c7 c8"
        >
          <button
            class="c9 c10 c11"
            data-testid="toggler-active"
            size="2"
            type="button"
          >
            <p
              class="c12 c13"
            >
              [value] module.settings.offLabel
            </p>
          </button>
          <button
            class="c14 c15 c11"
            data-testid="toggler-inactive"
            size="2"
            type="button"
          >
            <p
              class="c12 c13"
            >
              [value] module.settings.onLabel
            </p>
          </button>
        </section>
      </div>
    </div>
  </section>
</div>
`;
